src/DataObjects/QueuedJobDescriptor.php

```diff
@@ -585,4 +619,12 @@ private function getWorkerExpiry()
 
         return $time->getTimestamp() - $now->getTimestamp();
     }
+
+    public function scaffoldSearchFields($_params = null)
+    {
+        $fields = parent::scaffoldSearchFields($_params);
+        $fields->push($this->buildJobStatusField()->setEmptyString(''));
+        $fields->push($this->buildJobTypeField()->setEmptyString(''));
```

Comment:
Does this not result in duplicated fields? I'd have thought fields for these columns would already be scaffolded (so `$fields->replaceField()` would be appropriate)>

Reply:
FieldList doesn't let you have two fields with the same name. So pushing a field with a name that is already present in the fieldlist remove the pre-existing one. I think `replaceField` implicitly allows you to rename the field.

In this context, `push` seems cleaner to me but it's more of an aesthetic preference.

Reply:
> I think replaceField implicitly allows you to rename the field.

It literally replaces the field outright. Yes, that does mean to can rename the field if you want to, but that's irrelevant to what's happening here.

`replaceField()` seems cleaner to be because that's what you're _actually_ doing - you're replacing the field which has already been scaffolded. But like you say since `push()` implicitly removes the existing field, it's a subjective aesthetic choice.

I'm not fussed either way - I leave the decision to you and to whoever ends up doing a proper review of this after the card has been brought into a sprint.

Reply:
I'll leave it as-is.